### PR TITLE
Aws sdk upgrade to v2

### PR DESCRIPTION
**Pull request by proby**

### CODE REVIEW

File: lib/simple_job/sqs_job_queue.rb [hunk 1]
```diff
@@ -138,6 +139,13 @@ def poll(options = {}, &block)
         always_sleep: false
       }.merge(options)
 
+      sqs_receive_options = {
+        visibility_timeout: options[:visibility_timeout],
+        idle_timeout: options[:idle_timeout],
+        queue_url: sqs_queue_url,
+        max_number_of_messages: 1
+      }
+
```
Comment (proby) [inline lib/simple_job/sqs_job_queue.rb]: important change 1

File: lib/simple_job/sqs_job_queue.rb [hunk 2]
```diff
@@ -167,7 +175,7 @@ def poll(options = {}, &block)
         current_start_milliseconds = get_milliseconds
         current_job_type = 'unknown'
         begin
-          sqs_queue.receive_messages(options) do |message|
+          sqs_queue.receive_message(sqs_receive_options).messages.each do |message|
```
Comment (proby) [inline lib/simple_job/sqs_job_queue.rb]: important change 2

File: lib/simple_job/sqs_job_queue.rb [hunk 3]
```diff
@@ -207,6 +215,7 @@ def poll(options = {}, &block)
               # in child
               begin
                 message_handler.call(last_definition, last_message)
+                delete_message(last_message)
```
Comment (proby) [inline lib/simple_job/sqs_job_queue.rb]: important change 3

File: lib/simple_job/sqs_job_queue.rb [hunk 4]
```diff
@@ -215,6 +224,7 @@ def poll(options = {}, &block)
               exit # rubocop:disable Rails/Exit
             end
           else
+            delete_message(last_message)
```
Comment (proby) [inline lib/simple_job/sqs_job_queue.rb]: important change 4

File: lib/simple_job/sqs_job_queue.rb [hunk 5]
```diff
@@ -254,24 +262,31 @@ def poll(options = {}, &block)
       logger.info 'shutdown successful'
     end
 
+    def delete_message(message)
+      return unless message
+
+      sqs_queue.delete_message(queue_url: sqs_queue_url, receipt_handle: message.receipt_handle)
+    end
+
```
Comment (proby) [inline lib/simple_job/sqs_job_queue.rb]: important change 5

File: lib/simple_job/sqs_job_queue.rb [hunk 6]
```diff
@@ -234,13 +244,11 @@ def poll(options = {}, &block)
...
-          else
-            logger.error("unable to process message: #{e.message}")
-            logger.error("message body: #{last_message&.body}")
-            logger.error(e.backtrace.join("\n  "))
-          end
+          raise e if options[:raise_exceptions]
+
+          logger.error("unable to process message: #{e.message}")
+          logger.error("message body: #{last_message&.body}")
+          logger.error(e.backtrace.join("\n  "))
```
Comment (AlexWaffleman) [inline lib/simple_job/sqs_job_queue.rb:251]: think we need all this? maybe it doesn't happen all that much so it wouldn't be a huge deal.

File: lib/simple_job/sqs_job_queue.rb [hunk 7]
```diff
@@ -282,110 +297,114 @@ def get_milliseconds(time = Time.now.utc)
...
+        { name: 'Environment', value: self.class.config[:environment] },
+        { name: 'SQSQueueName', value: queue_name },
+        { name: 'Host', value: hostname }
+      ]
+
+      job_dimensions = message_dimensions + [
+        { name: 'JobType', value: job_type }
+      ]
+
+      metric_data = [
```
Comment (AlexWaffleman) [inline lib/simple_job/sqs_job_queue.rb:324]: this looks like just converting to new style hash syntax.

**Review by AlexWaffleman — APPROVED** [review]
just a couple of questions for my own info.

File: lib/simple_job/sqs_job_queue.rb (hunk 7 above)
Comment (proby) [reply lib/simple_job/sqs_job_queue.rb:324]: v1 keys were camel case and strings i.e. `'MetricName'`. v2 keys are snake case and symbols i.e. `metric_name`

Comment (AlexWaffleman) [reply lib/simple_job/sqs_job_queue.rb:324]: ah, yes. thanks.

File: lib/simple_job/sqs_job_queue.rb (hunk 6 above)
Comment (proby) [reply lib/simple_job/sqs_job_queue.rb:251]: That was there before, I drew an arbitrary line in how much refactoring to do ... and this didn't make the cut.

**Pull request merged**